server/main.lua

```diff
@@ -40,17 +40,35 @@ if Config.UseRPName then
     Framework.Object = nil -- free up the memory
 end
 
-local function getPlayerIdentifier(source)
-    local identifier
-    for _, v in ipairs(GetPlayerIdentifiers(source)) do
-        if string.match(v, "license:") then
-            identifier = string.gsub(v, "license:", "")
-            break
-        end
-    end
-    return identifier
+-- local function getPlayerIdentifier(source)
+    -- local identifier
+    -- for _, v in ipairs(GetPlayerIdentifiers(source)) do
+        -- if string.match(v, "license:") then
+            -- identifier = string.gsub(v, "license:", "")
+            -- break
+        -- end
+    -- end
+    -- return identifier
+-- end
+
+function GetPlayerIdentifierFromType(type, source)
+	local identifiers = {}
+	local identifierCount = GetNumPlayerIdentifiers(source)
+
+	for a = 0, identifierCount do
+		table.insert(identifiers, GetPlayerIdentifier(source, a))
+	end
+
+	for b = 1, #identifiers do
+		if string.find(identifiers[b], type, 1) then
+			return identifiers[b]
+		end
+	end
+	return nil
```

Comment:
to avoid all these calculations and loops, use `GetPlayerIdentifierByType` native.